### PR TITLE
Performance improving for MKL-DNN Quantized FullyConnected

### PR DESCRIPTION
**Pull request by ciyongch**
## Description ##
The patch is mainly for improving the performance of MKL-DNN quantized FullyConnected.

@pengzhao-intel @TaoLv 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/nn/fully_connected-inl.h [hunk 1]
```diff
@@ -48,6 +48,12 @@ enum FullyConnectedOpResource {kTempSpace};
 enum FullyConnectedOpOutputs {kOut};
 }  // fullc
 
+namespace quantized_fullc {
```
Comment (pengzhao-intel) [inline src/operator/nn/fully_connected-inl.h]: quantized_fc?

File: src/operator/subgraph/mkldnn/mkldnn_fc.cc [hunk 2]
```diff
@@ -195,6 +195,13 @@ void SgMKLDNNFCOp::Forward(const OpContext &ctx,
   }
 
   MKLDNNFCForwardFullFeature(full_param_, ctx, fwd_.get(), new_inputs, new_req, out_data);
+
+  if (mkldnn_param.quantized && !mkldnn_param.enable_float_output) {
```
Comment (pengzhao-intel) [inline src/operator/subgraph/mkldnn/mkldnn_fc.cc]: add comments on why

**Comment by abhinavs95** [comment]
@mxnet-label-bot update [MKLDNN, Performance, pr-awaiting-test]

File: src/operator/nn/fully_connected-inl.h (hunk 1 above)
Comment (ciyongch) [reply src/operator/nn/fully_connected-inl.h]: just to align with fullc :)

File: src/operator/subgraph/mkldnn/mkldnn_fc.cc (hunk 2 above)
Comment (ciyongch) [reply src/operator/subgraph/mkldnn/mkldnn_fc.cc]: I think it's straightforward here, the OutMin and OutMax are only valid when the op is quantized and not generating fp32 output.

**Comment by pengzhao-intel** [comment]
@anirudh2290 @ZhennanQin @xinyu-intel to review

File: src/operator/quantization/mkldnn/mkldnn_quantized_fully_connected.cc [hunk 3]
```diff
@@ -52,15 +47,15 @@ void MKLDNNQuantizedFullyConnectedForward(const nnvm::NodeAttrs &attrs,
   NDArray weight = in_data[fullc::kWeight];
 
   const float min_data =
-    in_data[num_inputs + quantized_fc_enum::kDataMin].data().dptr<float>()[0];
+    in_data[num_inputs + quantized_fullc::kDataMin].data().dptr<float>()[0];
```
Comment (ZhennanQin) [inline src/operator/quantization/mkldnn/mkldnn_quantized_fully_connected.cc]: Quite strange usage. Why not define a whole input sets with original inputs?

Comment (ciyongch) [reply src/operator/quantization/mkldnn/mkldnn_quantized_fully_connected.cc]: Original inputs might not include bias, which results in different index for all these min/max. Just to simplify the ordering for quantized op only.

**Comment by pengzhao-intel** [comment]
Thanks for your contribution. Merging now.
